### PR TITLE
Bug 1970331: pkg/securitycontextconstraints: wait for MCS/UID labels on namespaces, bump verbosity on errors

### PR DESCRIPTION
**Pull request by s-urbaniak**
/cc @stlaz @deads2k

### CODE REVIEW

File: pkg/securitycontextconstraints/sccmatching/matcher.go [hunk 1]
```diff
@@ -160,10 +161,37 @@ func constraintSupportsGroup(group string, constraintGroups []string) bool {
...
+			return false, nil
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		if lastErr != nil {
+			return nil, lastErr
+		}
+		return nil, err
```
Comment (stlaz) [inline pkg/securitycontextconstraints/sccmatching/matcher.go:192]: this will return a rather generic timeout error, you may want to customize it a bit

Comment (s-urbaniak) [reply pkg/securitycontextconstraints/sccmatching/matcher.go]: fixed

File: pkg/securitycontextconstraints/sccadmission/admission_test.go [hunk 2]
```diff
@@ -673,33 +674,38 @@ func TestCreateProvidersFromConstraints(t *testing.T) {
...
+			defer cancel()
+			_, errs := sccmatching.CreateProvidersFromConstraints(ctx, attributes.GetNamespace(), []*securityv1.SecurityContextConstraints{scc}, tc)
+
+			if !reflect.DeepEqual(scc, v.scc()) {
+				diff := diff.ObjectDiff(scc, v.scc())
+				t.Errorf("%s createProvidersFromConstraints mutated constraints. diff:\n%s", k, diff)
+			}
+			if len(v.expectedErr) > 0 && len(errs) != 1 {
+				t.Errorf("%s expected a single error '%s' but received %v", k, v.expectedErr, errs)
+				return
```
Comment (stlaz) [inline pkg/securitycontextconstraints/sccadmission/admission_test.go:695]: `t.Fatalf()` here and below with `t.Errorf` followed by a return

File: pkg/securitycontextconstraints/sccmatching/matcher.go [hunk 3]
```diff
@@ -196,21 +224,12 @@ func CreateProvidersFromConstraints(ns string, sccs []*securityv1.SecurityContex
...
 
 // CreateProviderFromConstraint creates a SecurityContextConstraintProvider from a SecurityContextConstraint
-func CreateProviderFromConstraint(ns string, namespace *corev1.Namespace, constraint *securityv1.SecurityContextConstraints, client kubernetes.Interface) (SecurityContextConstraintsProvider, *corev1.Namespace, error) {
+func CreateProviderFromConstraint(namespace *corev1.Namespace, constraint *securityv1.SecurityContextConstraints) (SecurityContextConstraintsProvider, error) {
 	var err error
 	resolveUIDRange := requiresPreAllocatedUIDRange(constraint)
 	resolveSELinuxLevel := requiresPreAllocatedSELinuxLevel(constraint)
 	resolveFSGroup := requiresPreallocatedFSGroup(constraint)
 	resolveSupplementalGroups := requiresPreallocatedSupplementalGroups(constraint)
-	requiresNamespaceAllocations := resolveUIDRange || resolveSELinuxLevel || resolveFSGroup || resolveSupplementalGroups
```
Comment (stlaz) [inline pkg/securitycontextconstraints/sccmatching/matcher.go:205]: nit: we could move the above functions to the place of use now

File: pkg/securitycontextconstraints/sccadmission/admission_test.go (hunk 2 above)
Comment (s-urbaniak) [reply pkg/securitycontextconstraints/sccadmission/admission_test.go:695]: fixed

File: pkg/securitycontextconstraints/sccmatching/matcher.go (hunk 3 above)
Comment (s-urbaniak) [reply pkg/securitycontextconstraints/sccmatching/matcher.go:205]: fixed

**Pull request merged**